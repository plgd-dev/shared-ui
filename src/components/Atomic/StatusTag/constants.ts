export const tagVariants = {
    SUCCESS: 'success',
    WARNING: 'warning',
    ERROR: 'error',
    NORMAL: 'normal',
    INFO: 'info',
} as const
