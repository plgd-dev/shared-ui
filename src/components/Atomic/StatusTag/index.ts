export { default } from './StatusTag'
export * from './StatusTag'
