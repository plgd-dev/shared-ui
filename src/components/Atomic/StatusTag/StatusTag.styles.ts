import { css } from '@emotion/react'
import { TagTypeType } from './StatusTag.types'
import { tagVariants } from './constants'
import { ThemeType, get } from '../_theme'

const getColorByVariant = (theme: ThemeType, variant: TagTypeType) => {
    if (variant === tagVariants.SUCCESS) {
        return css`
            color: ${get(theme, 'StatusTag.success.color')};
            background: ${get(theme, 'StatusTag.success.background')};
        `
    } else if (variant === tagVariants.WARNING) {
        return css`
            color: ${get(theme, 'StatusTag.warning.color')};
            background: ${get(theme, 'StatusTag.warning.background')};
        `
    } else if (variant === tagVariants.ERROR) {
        return css`
            color: ${get(theme, 'StatusTag.error.color')};
            background: ${get(theme, 'StatusTag.error.background')};
        `
    } else if (variant === tagVariants.NORMAL) {
        return css`
            color: ${get(theme, 'StatusTag.normal.color')};
            background: ${get(theme, 'StatusTag.normal.background')};
        `
    }
}

export const tag = (theme: ThemeType, variant: TagTypeType) => css`
    display: inline-flex;
    flex-direction: row;
    align-items: center;
    padding: 1px 10px;
    gap: 10px;
    border-radius: 44px;
    ${getColorByVariant(theme, variant)};
    font-family: 'Poppins', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 18px;
    text-transform: lowercase;
`
