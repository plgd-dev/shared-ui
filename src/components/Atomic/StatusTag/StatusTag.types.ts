import { ReactNode } from 'react'
import { tagVariants } from './constants'

export type TagTypeType = typeof tagVariants[keyof typeof tagVariants]

export type Props = {
    children: ReactNode
    onClick?: () => void
    variant: TagTypeType
}

export const defaultProps = {
    variant: tagVariants.SUCCESS,
}
