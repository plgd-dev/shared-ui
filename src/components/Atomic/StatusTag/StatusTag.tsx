import { FC } from 'react'
import { Props, defaultProps } from './StatusTag.types'
import * as styles from './StatusTag.styles'
import { tagSizes } from './constants'

const StatusTag: FC<Props> = (props) => {
    const { children, lowercase, onClick, size, variant } = { ...defaultProps, ...props }
    return (
        <div css={(theme) => [styles.tag(theme, variant), lowercase && styles.lowercase, size === tagSizes.MEDIUM && styles.mediumSize]} onClick={onClick}>
            {children}
        </div>
    )
}

StatusTag.displayName = 'StatusTag'

export default StatusTag
