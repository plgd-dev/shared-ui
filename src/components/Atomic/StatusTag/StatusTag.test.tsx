import React from 'react'
import { render } from '../../../../test/jest-wrapper'
import StatusTag from './StatusTag'

describe('<StatusTag>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <StatusTag variant='success'>status</StatusTag>
                <br />
                <br />
                <StatusTag variant='warning'>status</StatusTag>
                <br />
                <br />
                <StatusTag variant='error'>status</StatusTag>
                <br />
                <br />
                <StatusTag variant='normal'>status</StatusTag>
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
