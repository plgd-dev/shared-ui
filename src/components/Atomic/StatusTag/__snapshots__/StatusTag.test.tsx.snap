// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusTag> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1px 10px;
  gap: 10px;
  border-radius: 44px;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 18px;
  text-transform: lowercase;
}

<div>
    <div
      class="emotion-0"
    >
      status
    </div>
    <br />
    <br />
    <div
      class="emotion-0"
    >
      status
    </div>
    <br />
    <br />
    <div
      class="emotion-0"
    >
      status
    </div>
    <br />
    <br />
    <div
      class="emotion-0"
    >
      status
    </div>
  </div>
</DocumentFragment>
`;
